### PR TITLE
Regenerating preview SDK based off of latest swagger spec update

### PR DESCRIPTION
**Pull request by arv100kri**
## Description

1. Regenerate the preview SDK off of the changes to the swagger specifications from [this PR](https://github.com/Azure/azure-rest-api-specs/pull/5132)
2. Updated a skillset test to account for stricter validation in `DefaultLanguageCode` parameter for `OcrSkill`
3. Updated `DataSourceTests` to also account for stricter validation
4. Renamespace `SearchManagementClientTests`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by bitsforthought — CHANGES_REQUESTED** [review]
We need to discuss potential backcompat concerns before I can merge this. Otherwise, the changes look good.

**Comment by brjohnstmsft** [comment]
I missed it in my original review, but could you please also regenerate the code for `Microsoft.Azure.Search.Data`?

**Comment by arv100kri** [comment]
yup, I've done that. Commit [797faef9823b453f4130d3ac4288f047bb84c8a5](https://github.com/Azure/azure-sdk-for-net/pull/5203/commits/797faef9823b453f4130d3ac4288f047bb84c8a5)

---
In reply to: [459909441](https://github.com/Azure/azure-sdk-for-net/pull/5203#issuecomment-459909441) [](ancestors = 459909441)

File: src/SDKs/Search/DataPlane/Search.Tests/Tests/SkillsetsTests.cs
Comment (arv100kri) [reply src/SDKs/Search/DataPlane/Search.Tests/Tests/SkillsetsTests.cs]: We discussed that this is a 'reasonable' exemption.

**Review by bitsforthought — CHANGES_REQUESTED** [review]
Had a question about a credential in a test; Otherwise LGTM.

File: src/SDKs/Search/DataPlane/Search.Tests/Tests/DataSourceTests.cs
Comment (arv100kri) [reply src/SDKs/Search/DataPlane/Search.Tests/Tests/DataSourceTests.cs]: This is a "fake" credential... It just conforms to the "format" of a connection string..
We've recently added validation logic to connection strings, and this is to get past that

**Review by bitsforthought — APPROVED** [review]
@arv100kri Changes look good. We just need to wait for your Swagger change to be merged so you can regenerate one more time to update the metadata, and then I'll merge this.

**Comment by brjohnstmsft** [comment]
@arv100kri Apparently I'm a bit behind. I see the Swagger has been merged, and the metadata indicates that the regenerated code was based off Azure/master. I'm going to merge this now.

**Pull request merged**
